app/src/main/java/org/mozilla/fenix/library/history/HistoryStore.kt

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.history
+
+import mozilla.components.lib.state.Action
+import mozilla.components.lib.state.State
+import mozilla.components.lib.state.Store
+
+/**
+ * Class representing a history entry
+ * @property id Unique id of the history item
+ * @property title Title of the history item
+ * @property url URL of the history item
+ * @property visitedAt Timestamp of when this history item was visited
+ */
+data class HistoryItem(val id: Int, val title: String, val url: String, val visitedAt: Long)
+
+/**
+ * An alias to make it easier to work with `Store<HistoryState, HistoryAction>`
+ */
+typealias HistoryStore = Store<HistoryState, HistoryAction>
```

Comment:
@ekager We've landed an update to the `SearchStore` which we should apply here as well. Basically, turn this into a class and passing the reducer directly. This makes it easier to read and also prevents a problem where the method reference potentially captures scope it shouldn't be and causing leaks:

https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/search/SearchStore.kt#L16